arelle/plugin/validate/ESEF/ESEF_Current/DTS.py

```diff
@@ -109,7 +118,13 @@ def lc3wordAdjust(word: str) -> str:
                         val.modelXbrl.error("ESEF.3.4.3.extensionTaxonomyDimensionNotAssignedDefaultMemberInDedicatedPlaceholder",
                             _("Each dimension in an issuer specific extension taxonomy MUST be assigned to a default member in the ELR with role URI http://www.esma.europa.eu/xbrl/role/core/ifrs-dim_role-990000 defined in esef_cor.xsd schema file. %(qname)s"),
                             modelObject=modelConcept, qname=modelConcept.qname)
-                    esefDomainItemTypes = qnDomainItemTypes if getDisclosureSystemYear(val.modelXbrl) < 2023 else qnDomainItemTypes2023
+
+                    if getDisclosureSystemYear(val.modelXbrl) < 2023:
```

Comment:
The disclosure system year value is being used so frequently now that it likely makes sense to assign it to a variable (such as `esefYear`) and reuse it to avoid repeated regex matches. Additionally, it would be beneficial to clarify the distinction between the two variables, `esefTaxonomyYear` and `esefDisclosureSystemYear`.